src/JNI/com/amazonaws/kinesis/video/producer/jni/Parameters.cpp

```diff
@@ -249,10 +251,176 @@ BOOL setClientInfo(JNIEnv *env, jobject clientInfo, PClientInfo pClientInfo) {
         CHK_JVM_EXCEPTION(env);
     }
 
+    methodId = env->GetMethodID(cls, "getMetricLoggingPeriod", "()J");
```

Comment:
I couldn't find this method in the Java SDK. Is there something that I'm missing here?

Reply:
+1. We need this and getReservedCallbackPeriod in your Java PR.

Reply:
These changes were intended to work to null-initialize the currently not initialized members without any changes being made in the Java end. This PR was not to add support for theses members and classes in Java, but to prevent undefined behavior from these members not being null-initialized. Will discuss more offline in case we do want this to support setting/getting members from Java.

Reply:
I understand, I'm not sure if it is a good idea to look for functions that don't exist in the Java layer. Can we initialize it without looking for that function instead? Because it is easy to misunderstand this as something missing

Reply:
That's what my changes were doing before we decided to add in all this framework to be able to readily hook it up to the Java end, and the Java-side changes were just for parity purposes just to be used as reference. It looks like we do want to somewhat implement this in the Java end now though, so I'm working on setting up Java getters that simply return null rather than the expected class object.

Reply:
Understood, thanks for taking care of this!